### ISSUE TITLE
Support pre-buffering of next concatenated item.

### ISSUE THREAD

**Issue by tonihei** [google/ExoPlayer#3327]
Currently items in a concatenation of media periods are buffered consecutively. This usually results in a buffering state when items are skipped by the user. To support such use-cases, it would be helpful to pre-buffer the start of the next period even if the current period hasn't been fully buffered yet.

**Comment by keerthikorrapati** [comment]
@tonihei  - This enhancement is already shipped right? Looks like we are using DefaultLoadControl to continue pre-buffering of the next period as well. Can you please let me know if this is what's happening :)

**Comment by tonihei** [comment]
No, sorry, this is still something we are planning to do but haven't got around to do it.

**Comment by ojw28** [comment]
To clarify:
* It is true that we already start to pre-buffer the next item(s) once the current item has been fully buffered.
* The enhancement being tracked here is to consider pre-buffering the start of the next item(s) in a playlist _before_ we've buffered all the way to the end of the current item. The reasoning being that users are more likely to skip to the start of the next item than seek toward the end of the current item. This is particularly true for music apps.

**Comment by keerthikorrapati** [comment]
Thank you for clarifying @ojw28 and @tonihei . While we are at it, I am building a DynamicConcatenatingMediaSource which can take up to 30 videos. I built a demo and saw that exoplayer pre-buffers only one or two next items. So is it okay to add this kind of load or should we code up a wrapper logic to keep up to only k videos at a time.

**Comment by tonihei** [comment]
ExoPlayer buffers as much media as configured in your LoadControl. That's independent of the length of the playlist. So, in short, it's fine to keep as many items in the playlist as you like.

**Comment by keerthikorrapati** [comment]
Thank you :)

**Comment by gracejang** [comment]
Hello! Wondering if work has been started yet to support this?

**Comment by tonihei** [comment]
Not yet, sorry. We'll keep this issue posted if we have any updates.

**Comment by ghost** [comment]
> Not yet, sorry. We'll keep this issue posted if we have any updates.

Any news about this issue ?

**Comment by nilskassube** [comment]
A different use case to consider: sometimes customers explicitly do not want to start buffering before a previous item has been fully played. This is useful for radio stations to ensure that streams do not have too much delay, i.e. for alarm clock intros or audio ads.

**Comment by Divere** [comment]
Issue up! Still needed this functionality.

**Comment by elkasvirElka** [comment]
I was searching for this opportunity also. Does anybody have their custom implementation?

**Comment by yangcheng** [comment]
is there a timeline for this request or a 3rd party implementation?

**Comment by anirudhramanan** [comment]
Any update here ? We are also trying to solve for a similar use-case i.e to pre-fetch some segments of all the media items in the playlist

**Comment by developerzjy** [comment]
I have implemented the pre-caching function, this is my demo [https://github.com/developerzjy/ExoDemo](https://github.com/developerzjy/ExoDemo)

**Comment by xshiwei** [comment]
Any news about this issue ?

**Comment by krupa-p-simformsolutions** [comment]
Any updates on this issue? Or anyone else finds any solution for the same?

**Comment by ryanmord** [comment]
Also looking for a solution to this issue
